.github/workflows/pull-request-tests.yaml

```diff
@@ -13,13 +13,13 @@ env:
   PYTHON_VERSION: 3.11
 
 jobs:
-  check_action:
+  check-action:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
-          python-version: "${{ env.PYTHON_VERSION }}"
+          python-version: ${{ env.PYTHON_VERSION }}
```

Comment:
Quotes are good practice here. One would not be able to test against Python 3.10 because Yaml interprets that as float and makes it 3.1. While that's a theoretical problem here, I always advocate to quote number-like strings.

Reply:
Ah, this was done with the expectation of a potential Python version with zero at the end sometime in the future. OK, I may revert this.

Reply:
Hi @renefritze By the way, I am currently testing further measures to tighten the security of this Action, and I should say that, when GitHub-specific expressions `${{ ... }}` are in use, no YAML-specific conversion performed for scalars (no trailing zero removal for "floats", no "hexadecimal" or "octal" values, etc). For example, in case like this:

```yaml
    - uses: actions/checkout@v4
      with:
        repository: ${{ env.PR_HEAD_REPO }}
        ref: ${{ env.PR_HEAD_SHA }}
        persist-credentials: false
```

if `PR_HEAD_SHA` is set to 2.10, trailing zero is [not removed](https://github.com/oleg-derevenetz/fheroes2/actions/runs/7701418438/job/20987411270#step:5:19) even though there are no quotes. The same is when `PR_HEAD_SHA` is set to 0x11 - this value is [not interpreted](https://github.com/oleg-derevenetz/fheroes2/actions/runs/7701902904/job/20989005725#step:5:19) as "hexadecimal" value. Apparently, all these substitutions happen somewhere inside the GitHub Actions Runner after the YAML parser has done its job, and thus no additional quotes are required.

Reply:
Ahh, OK, thanks for the heads up.